pallets/bucket-nfts/src/tests.rs

```diff
@@ -0,0 +1,369 @@
+use frame_support::{assert_noop, assert_ok};
+use sp_core::{ByteArray, Hasher};
+use sp_keyring::sr25519::Keyring;
+use sp_runtime::BoundedVec;
+use storage_hub_traits::ReadProvidersInterface;
+
+use crate::{
+    mock::{new_test_ext, BucketNfts, FileSystem, RuntimeOrigin, System, Test},
+    types::{ItemMetadata, ReadAccessRegex},
+    Error, Event,
+};
+
+mod share_access_tests {
```

Comment:
Add a test for a bucket that was originally private, then public and then try to share access.